esphome/components/ld2410s/ld2410s.cpp

```diff
@@ -0,0 +1,1146 @@
+  this->min_dist_ = static_cast<float>(min_distance) / 0.7f;
+  this->schedule_cmd_("set_min_distance\0", PARAMS_WRITE_CMD, CFG_MIN_DETECTION_VALUE);
+}
+void LD2410S::set_status_reporting_freq(float status_reporting_freq) {
+  this->status_freq_ = status_reporting_freq * 10;
+  this->schedule_cmd_("set_status_reporting_freq\0", PARAMS_WRITE_CMD, CFG_STATUS_FREQ_VALUE);
+}
+void LD2410S::set_threshold_selected_gate(float threshold_selected_gate) {
+  this->thresholds_.selected_gate = threshold_selected_gate;
+#ifdef USE_NUMBER
+  this->threshold_selected_gate_number_->publish_state(this->thresholds_.selected_gate);
+  this->threshold_trigger_number_->publish_state(this->thresholds_.trigger[this->thresholds_.selected_gate]);
+  this->threshold_hold_number_->publish_state(this->thresholds_.hold[this->thresholds_.selected_gate]);
+  this->threshold_snr_number_->publish_state(this->thresholds_.snr[this->thresholds_.selected_gate]);
+#endif
+}
+void LD2410S::set_threshold_trigger(float threshold_trigger) {
+  this->thresholds_.trigger[this->thresholds_.selected_gate] = threshold_trigger;
+  this->schedule_cmd_("set_threshold_trigger\0", GATE_THRESHOLD_TRIGGER_WRITE_CMD, this->thresholds_.selected_gate);
+  this->publish_threshold_trigger_();
+}
+void LD2410S::set_threshold_hold(float threshold_hold) {
+  this->thresholds_.hold[this->thresholds_.selected_gate] = threshold_hold;
+  this->schedule_cmd_("set_threshold_hold\0", GATE_THRESHOLD_HOLD_WRITE_CMD, this->thresholds_.selected_gate);
+  this->publish_threshold_hold_();
+}
+void LD2410S::set_threshold_snr(float threshold_snr) {
+  this->thresholds_.snr[this->thresholds_.selected_gate] = threshold_snr;
+  this->schedule_cmd_("set_threshold_snr\0", GATE_THRESHOLD_SNR_WRITE_CMD, this->thresholds_.selected_gate);
+  this->publish_threshold_snr_();
+}
+void LD2410S::set_response_speed_select(const std::string &response_speed_select) {
+  this->resp_speed_ = response_speed_select == RESPONSE_SPEED_NORMAL ? 5 : 10;
+  this->schedule_cmd_("set_response_speed_select\0", PARAMS_WRITE_CMD, CFG_RESPONSE_SPEED_VALUE);
+}
+
+void LD2410S::schedule_cmd_(const char *msg, uint16_t command, uint16_t sub_command) {
+  this->status_set_warning(msg);
+
+  this->schedule_cmd_frame_(CONFIG_MODE_START_CMD);
+  this->schedule_cmd_frame_(command, sub_command);
+  this->schedule_cmd_frame_(CONFIG_MODE_END_CMD);
+
+  this->status_clear_warning();
+}
+
+void LD2410S::schedule_cmd_frame_(uint16_t command, uint16_t sub_command) {
+  ESP_LOGD(TAG, "schedule_cmd_frame %x : %x", command, sub_command);
+
+  CmdFrameT cmd_frame = {.header = CMD_FRAME_HEADER, .footer = CMD_FRAME_FOOTER, .command = command, .data_length = 0};
+
+  switch (command) {
+    case OUTPUT_MODE_SWITCH_CMD: {
+      if (this->minimal_output_) {
+        this->cmd_frame_append_data_(&cmd_frame, &OUTPUT_MODE_VALUE_MIN[0], 4);
+      } else {
+        this->cmd_frame_append_data_(&cmd_frame, &OUTPUT_MODE_VALUE_STD[0], 4);
```

Comment:
```suggestion
        this->cmd_frame_append_data_(&cmd_frame, &OUTPUT_MODE_VALUE_MIN[0], 6);
      } else {
        this->cmd_frame_append_data_(&cmd_frame, &OUTPUT_MODE_VALUE_STD[0], 6);
```
`OUTPUT_MODE_VALUE_MIN` & `OUTPUT_MODE_VALUE_STD` are 6 not 4 bytes. Seems sending the 4 bytes you are twice still works, but I *think* sending all 6 bytes at once is better than timing out and re-sending.